File: model/src/sinus/SinusReader.java
package sinus;

import data.DataReader;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class SinusReader implements DataReader<Double> {

    private final double periodsCount;
    private final int size;

    public SinusReader(double periodsCount, int size) {
        this.periodsCount = periodsCount;
        this.size = size;
    }

    @Override
    public List<Double> readData() throws IOException {
        List<Double> resultData = new ArrayList<>();
        double step = Math.PI * 2 * periodsCount / size;
        for (double value = 0; resultData.size() < size; value += step) {
            resultData.add(Math.sin(value));
        }
        return resultData;
    }
}

File: model/src/sinus/SinusReader.kt
package sinus

import data.DataReader
import java.io.IOException
import java.util.stream.DoubleStream

class SinusReader(private val periodsCount: Double, private val size: Int) : DataReader<Double> {

    @Throws(IOException::class)
    override fun readData(): List<Double> {
        val step = Math.PI * 2.0 * periodsCount / size
        return DoubleStream.iterate(0.0, { i -> i + step }).map { x -> Math.sin(x) }.limit(size.toLong()).toArray().asList()
    }
}
